hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterSafemodeService.java

```diff
@@ -174,7 +167,7 @@ public void periodicInvoke() {
     // Always update to indicate our cache was updated
     if (isCacheStale) {
       if (!safeMode) {
-        enter();
+        enter(false);
```

Comment:
Wouldn't make more sense to use `leave()`?
It would be good to have unified enter and leave, and just have a call to each.